File: Beecrowd_Parte1/Banco_do_Farao_2071.java
import java.util.Scanner;

public class Banco_do_Farao_2071 {

    // Função para calcular a soma de um subarray máximo e o seu tamanho
    public static void encontrarMaximoSubarray(int[] contas, int inicio, int fim, Result result) {
        int somaAtual = 0;
        int melhorSoma = Integer.MIN_VALUE;
        int comprimentoAtual = 0;
        int melhorComprimento = 0;

        for (int i = inicio; i <= fim; i++) {
            somaAtual += contas[i];
            comprimentoAtual++;

            if (somaAtual > melhorSoma || (somaAtual == melhorSoma && comprimentoAtual > melhorComprimento)) {
                melhorSoma = somaAtual;
                melhorComprimento = comprimentoAtual;
            }

            if (somaAtual < 0) {
                somaAtual = 0;
                comprimentoAtual = 0;
            }
        }

        result.somaMaxima = melhorSoma;
        result.comprimentoMaximo = melhorComprimento;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int T = scanner.nextInt();

        while (T-- > 0) {
            int N = scanner.nextInt();
            int[] contas = new int[N + 1];

            for (int i = 1; i <= N; i++) {
                contas[i] = scanner.nextInt();
            }

            int Q = scanner.nextInt();

            for (int i = 0; i < Q; i++) {
                int A = scanner.nextInt();
                int B = scanner.nextInt();

                Result result = new Result();
                encontrarMaximoSubarray(contas, A, B, result);
                System.out.println(result.somaMaxima + " " + result.comprimentoMaximo);
            }
        }

        scanner.close();
    }

    // Classe para armazenar os resultados da função encontrarMaximoSubarray
    static class Result {
        int somaMaxima;
        int comprimentoMaximo;
    }
}

File: Beecrowd_Parte1/Banco_do_Farao_2071.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

// Função para calcular a soma de um subarray máximo e o seu tamanho
void encontrar_maximo_subarray(int* contas, int inicio, int fim, int* soma_maxima, int* comprimento_maximo) {
    int soma_atual = 0;
    int melhor_soma = INT_MIN;
    int comprimento_atual = 0;
    int melhor_comprimento = 0;

    for (int i = inicio; i <= fim; i++) {
        soma_atual += contas[i];
        comprimento_atual++;

        if (soma_atual > melhor_soma || (soma_atual == melhor_soma && comprimento_atual > melhor_comprimento)) {
            melhor_soma = soma_atual;
            melhor_comprimento = comprimento_atual;
        }

        if (soma_atual < 0) {
            soma_atual = 0;
            comprimento_atual = 0;
        }
    }

    *soma_maxima = melhor_soma;
    *comprimento_maximo = melhor_comprimento;
}

int main() {
    int T;
    scanf("%d", &T);

    while (T--) {
        int N;
        scanf("%d", &N);
        int* contas = (int*)malloc((N + 1) * sizeof(int));
        
        for (int i = 1; i <= N; i++) {
            scanf("%d", &contas[i]);
        }

        int Q;
        scanf("%d", &Q);

        for (int i = 0; i < Q; i++) {
            int A, B;
            scanf("%d %d", &A, &B);

            int soma_maxima, comprimento_maximo;
            encontrar_maximo_subarray(contas, A, B, &soma_maxima, &comprimento_maximo);
            printf("%d %d\n", soma_maxima, comprimento_maximo);
        }

        free(contas);
    }

    return 0;
}
